File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/youtube/service/SubscriptionUpdateSheduler.java
package com.github.bumblebee.command.youtube.service;

import com.github.bumblebee.command.youtube.entity.Subscription;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import java.util.Date;
import java.util.concurrent.TimeUnit;

@Component
public class SubscriptionUpdateSheduler {

    private final YoutubeSubscriptionService service;
    private static final long delay = 60 * 1000 * 60 * 4; //4 Hours in millis
    private static final long overdueInterval = TimeUnit.DAYS.toMillis(4);

    @Autowired
    public SubscriptionUpdateSheduler(YoutubeSubscriptionService service) {
        this.service = service;
    }

    @Scheduled(fixedRate = delay)
    public void checkOverdueSubscriptions() {
        Date date = new Date();
        for (Subscription sub : service.getExistingSubscriptions()) {
            long interval = date.getTime() - sub.getUpdatedDate().getTime();
            if (interval > overdueInterval) {
                sub.setUpdatedDate(date);
                if (service.subscribeChannel(sub.getChannelId()))
                    service.storeSubscription(sub);
            }
        }
    }

}

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/youtube/service/SubscriptionUpdateSheduler.kt
package com.github.bumblebee.command.youtube.service

import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import java.util.*
import java.util.concurrent.TimeUnit

@Component
class SubscriptionUpdateSheduler(private val service: YoutubeSubscriptionService) {

    @Scheduled(fixedRate = delay)
    fun checkOverdueSubscriptions() {
        val date = Date()
        for (sub in service.existingSubscriptions) {
            val interval = date.time - sub.updatedDate.time
            if (interval > overdueInterval) {
                sub.updatedDate = date
                if (service.subscribeChannel(sub.channelId))
                    service.storeSubscription(sub)
            }
        }
    }

    companion object {
        private const val delay = (60 * 1000 * 60 * 4).toLong() //4 Hours in millis
        private val overdueInterval = TimeUnit.DAYS.toMillis(4)
    }

}
